src/main/java/Script.java

```diff
@@ -0,0 +1,107 @@
+import org.json.JSONArray;
+import java.util.Map;
+/**
+ * @author B. Clint Hall
+ * @version 1
+ */
+public class Script {
+    private final static int INDEX_OF_BLOCK_TUPLE_ARRAY_IN_SCRIPT_TUPLE = 2;
+    private JSONArray blocks = null;
+
+    /**
+     * Constructs a script from a script tuple.
+     * @param scriptTuple A JSONArray containing an X coordinate, a Y coordinate
+     *      and a JSONArray of blocks.
+     */
+    public Script(JSONArray scriptTuple) {
+
+        if (scriptTuple != null) {
+            blocks = scriptTuple.optJSONArray(INDEX_OF_BLOCK_TUPLE_ARRAY_IN_SCRIPT_TUPLE);
+        }
+        if (blocks == null) {
+            blocks = new JSONArray();
+        }
+    }
+    /**
+     * Counts the number of blocks of each type for this script
+     * of block tuples.
+     * @return an integer array in which each int is a category count.  Uses teh order
+     * of {@code ScriptSpecs.getCommandsByType}.
+     */
+    public int[] sumBlocksByCategory() {
+        int[] blocksByCategory = new int[ScriptSpecs.getCategories().length];
+        sumBlocksByCategory(blocksByCategory);
+        return blocksByCategory;
+    }
+    /**
+     * Counts the number of blocks of each type for this script
+     * of block tuples.
+     * @param blocksByCategory an integer array. Uses the order
+     * of {@code ScriptSpecs.getCommandsByType}. Integer at each index
+     * will be incremented for each block of corresponding type.
+     */
+    public void sumBlocksByCategory(int[] blocksByCategory) {
+        sumBlocksByCategory(blocks, blocksByCategory);
+    }
+    /**
+     * Get the number of blocks in the script.
+     * @return the number of blocks in the script.
+     */
+    public int getLength() {
+        return blocks.length();
+    }
+    /**
+     * Counts the number of blocks of each type in a JSONArray
+     * of block tuples and adds them into {@code blocksByCategory}.
+     * @param blocks A JSONArray of JSONArrays.  This is how scripts are
+     *          represented in the project.json files inside .sb2 files.
+     * @param blocksByCategory The array in which to increment block
+     * counts by type.
+     */
+    private static void sumBlocksByCategory(JSONArray blocks, int[] blocksByCategory) {
```

Comment:
@whereswaldon we need the `blocks` parameter because we call it recursively through `handleNestedBlocks`.  It doesn't have to be static, but I decided to make it static to signal that it doesn't alter state.